### ISSUE TITLE
setAudioSource Failed on SoundRecorder.stop()

### ISSUE THREAD

**Issue by ameenmattar** [kevinresol/react-native-sound-recorder#10]
Thank you for this amazing library @kevinresol  , but i am facing some problems:
The app crashes when  on SoundRecorder.stop() 
works fine on _**iOS**_  but crashes on **_Android_**, 
i have included the android permission in the AndroidManifest as you mentioned : 
`    <uses-permission android:name="android.permission.RECORD_AUDIO" />`

**And here is the code for SoundRecorder.stop():** 

```
SoundRecorder.stop().then(function(result) {
      console.log("stopped recording, audio file saved at: ", result.path);
      this.onSendFile(result.path, "test.mp4", "audio/mpeg");
    });
```

**The Error i get is described in the below image:** 

![screen shot 2018-07-19 at 11 25 33 am](https://user-images.githubusercontent.com/32841332/42931336-1dbf93b8-8b48-11e8-8552-dd136529b037.png)

Thank you in advance

**Comment by kevinresol** [comment]
~~same as https://github.com/kevinresol/react-native-sound-recorder/issues/8?~~

well no, #8 crashes at `start()`

**Comment by kevinresol** [comment]
but that doesn't make sense. setAudioSource only happens in start()

**Comment by kevinresol** [comment]
did you request for the permission at runtime?

**Comment by ameenmattar** [comment]
@kevinresol  Yes i have requested for the permission at runtime 
and added this permission to the AndroidManifest : 

` <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />`

**Comment by kevinresol** [comment]
Can you show the code for the run time permission request?

**Comment by ameenmattar** [comment]
Here's my run time permission request code: 
```
async function requestMicrophonePermission() {
  try {
    const granted = await PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.RECORD_AUDIO,
      {
        title: "Requesting Microphone Access",
        message: "App needs permission to access to your microphone "
      }
    );
    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
      console.log("You can use the microphone");
    } else {
      console.log("Microphone permission denied");
      alert("Microphone permission denied");
    }
  } catch (err) {
    console.warn(err);
  }
}
```

**Comment by kevinresol** [comment]
> but that doesn't make sense. setAudioSource only happens in start()

are you sure the crash happens at `stop()`?

**Comment by kevinresol** [comment]
What is your android version and phone model btw?

**Comment by ameenmattar** [comment]
Yes it seems to happen on stop(), and even if its on start() #8  didn't solve my problem because i have added the permission and nothing changed.
The android version "27.0.3"
The phone model : Samsung S8

**Comment by kevinresol** [comment]
Can you try `<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />`
and request the permission at runtime too?

**Comment by ameenmattar** [comment]
I have already include these too 

_**ANDROID_STUDIO :**_

![screen shot 2018-07-20 at 5 43 06 pm](https://user-images.githubusercontent.com/32841332/43008804-a599f5ce-8c44-11e8-958c-2d8353635e79.png)

**_JS_CODE:_**

![screen shot 2018-07-20 at 5 43 26 pm](https://user-images.githubusercontent.com/32841332/43008833-b8886562-8c44-11e8-890e-224557813226.png)

**And here is the Code for Recording:**

```
async _onMicroPress() {
    await requestMicrophonePermission();
    await requestStoragePermission();
    SoundRecorder.start(SoundRecorder.PATH_CACHE + "/test.mp4").then(
      function() {
        console.log("started recording");
      }
    );
  }

  _onMicroRelease = () => {
    SoundRecorder.stop().then(function(result) {
      console.log("stopped recording, audio file saved at: ", result.path);
      this.onSendFile(result.path, "test.mp4", "audio/mpeg");
    });
  };
```

**Comment by kevinresol** [comment]
To be honest I don't have any idea at the moment

**Comment by geniuscd** [comment]
Hey, Thanks for the support. I found a fix for this issue. basically its something related to the android only and the "MediaRecorder" class. PR #11

**Comment by kevinresol** [comment]
@ameenmattar can you try again with v1.3.1 (merged #11)?

**Comment by kevinresol** [comment]
I assume it is fixed
